pallets/contracts/src/lib.rs

```diff
@@ -251,16 +251,16 @@ decl_module! {
             with_transaction(|| {
                 // Call underlying function
                 <pallet_contracts::Module<T>>::put_code(origin, code)?;
-                // Update the storage.
-                <TemplateInfo<T>>::insert(code_hash, TemplateDetails {
-                    instantiation_fee,
-                    owner: did,
-                    frozen: false
-                });
-                <MetadataOfTemplate<T>>::insert(code_hash, meta_info);
                 // Charge the protocol fee
                 T::ProtocolFee::charge_fee(ProtocolOp::ContractsPutCode)
             })?;
+            // Update the storage.
```

Comment:
I doubt this would show up either way in benchmarks outside of noise. The impact is going to be quite small.

From reading the code below, my understanding of transactions is that buffers are recorded in Rust and bubbled up... Nothing actually touches disk until we do root-level level transaction commits. With a more narrow `with_transaction`, you'll have fewer "moving on up" copies basically.

https://github.com/paritytech/substrate/blob/947a6bc1cc8919d382c193893308ba46e6d6cba7/primitives/state-machine/src/overlayed_changes/changeset.rs#L335-L377